### ISSUE TITLE
Metro Bundler Not Working on 0.60

### ISSUE THREAD

**Issue by mreorhan** [facebook/react-native#25509]
React Native version:
System:
    OS: Windows 10
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Memory: 4.22 GB / 15.89 GB
  Binaries:
    Node: 10.15.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.13.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.4.1 - C:\Program Files\nodejs\npm.CMD
  IDEs:
    Android Studio: Version  3.4.0.0 AI-183.6156.11.34.5522156

## Steps To Reproduce

1. react-native init AwesomeProject
2. react-native run-android

Describe what you expected to happen:
When I initialized a new project with 0.60, metro bundler not opening. Project starts successfully but app gave bundler error.

**Comment by dharmasat** [comment]
also happened to me from yesterday .it loads metro bundler but it close suddenly after invoking

**Comment by thymikee** [comment]
It's likely issue with Windows, as we didn't experience this in UNIX environment. Could you help us debug this? I don't have easy access to Windows machine at the moment.

**Comment by pablohenriq** [comment]
I can confirm on the Unix platform. I reproduced the above process in elementary OS 5.0 Juno (based on Ubuntu 18.04 LTS).

Running `react-native start` in another terminal section, apparently reaches a timeout, I said apparently because everything gets ridiculously slow when I try to run it.

**Comment by thymikee** [comment]
Let's move this discussion to the CLI repo: https://github.com/react-native-community/cli/issues/484

**Comment by thymikee** [comment]
We released a fix for that in `@react-native-community/cli@2.1.1`. Please [update the CLI](https://github.com/react-native-community/cli#updating-the-cli) and see if it works for you.

**Comment by Mahi-Uddin** [comment]
I am facing a weird issue.
When I do `react-native run-android`, the output is
```
 zihad@zihad-ubuntu  ~/Documents/ReactNative/Projects/AwesomeProject  react-native run-android
info Running jetifier to migrate libraries to AndroidX. You can disable it using "--no-jetifier" flag.
Jetifier found 855 file(s) to forward-jetify. Using 4 workers...
info Starting JS server...
info Installing the app...

> Task :app:installDebug
12:31:42 V/ddms: execute: running am get-config
12:31:42 V/ddms: execute 'am get-config' on '8f9ebf1' : EOF hit. Read: -1
12:31:42 V/ddms: execute: returning
Installing APK 'app-debug.apk' on 'POCOPHONE F1 - 9' for app:debug
12:31:42 D/app-debug.apk: Uploading app-debug.apk onto device '8f9ebf1'
12:31:42 D/Device: Uploading file onto device '8f9ebf1'
12:31:42 D/ddms: Reading file permision of /home/zihad/Documents/ReactNative/Projects/AwesomeProject/android/app/build/outputs/apk/debug/app-debug.apk as: rw-r--r--
12:31:42 V/ddms: execute: running pm install -r -t "/data/local/tmp/app-debug.apk"
12:31:46 V/ddms: execute 'pm install -r -t "/data/local/tmp/app-debug.apk"' on '8f9ebf1' : EOF hit. Read: -1
12:31:46 V/ddms: execute: returning
12:31:46 V/ddms: execute: running rm "/data/local/tmp/app-debug.apk"
12:31:46 V/ddms: execute 'rm "/data/local/tmp/app-debug.apk"' on '8f9ebf1' : EOF hit. Read: -1
12:31:46 V/ddms: execute: returning
Installed on 1 device.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 7s
27 actionable tasks: 2 executed, 25 up-to-date
info Connecting to the development server...
info Starting the app on "8f9ebf1"...
Starting: Intent { cmp=com.awesomeproject/.MainActivity }
```
Then it quits, I checked with `netstat -tulpn` it didn't start listening on port 8081.
And when I do `react-native start`, the output is
```
┌──────────────────────────────────────────────────────────────────────────────┐
│                                                                              │
│  Running Metro Bundler on port 8081.                                         │
│                                                                              │
│  Keep Metro running while developing on any JS projects. Feel free to        │
│  close this tab and run your own Metro instance if you prefer.               │
│                                                                              │
│  https://github.com/facebook/react-native                                    │
│                                                                              │
└──────────────────────────────────────────────────────────────────────────────┘

Looking for JS files in
   /home/zihad/Documents/ReactNative/Projects/AwesomeProject 

Loading dependency graph...
```
and it stays here forever.
Also, it starts listening on port 8081, but it freezes my laptop.
Then I rebooted it with the power button and tried with nothing open other than 2 terminal tabs, I ran `top` and saw that one of the `node` processes gradually takes up 100% of my CPU and after a while it freezes.
It doesn't happen on 0.59.10, only if I use 0.60.x
Hope I will get some help, TIA.

**Comment by thymikee** [comment]
@Mahi-Uddin downgrade your Node version to 12.4 or lower

**Comment by Mahi-Uddin** [comment]
That fixed the freezing issue, but still running `react-native run-android` says starting JS server, but doesn't actually start it.
I need to run `react-native start`, is there any way to fix it?
And also the screen is white with the template.
And `react-native log-android` only shows this, then nothing else.
```
info Starting logkitty
```

**Comment by Mahi-Uddin** [comment]
> That fixed the freezing issue, but still running `react-native run-android` says starting JS server, but doesn't actually start it.
> I need to run `react-native start`, is there any way to fix it?
> And also the screen is white with the template.
> And `react-native log-android` only shows this, then nothing else.
> 
> ```
> info Starting logkitty
> ```

Nothing on this?

**Comment by thymikee** [comment]
@Mahi-Uddin do you have a repro we can download?

**Comment by Mahi-Uddin** [comment]
The issue is reproducible with [this](https://github.com/Mahi-Uddin/firebase_todo) repo.
All I did is `react-native init firebase_todo` and `react-native run-android`, it didn't start listening on port 8081 but installed the app, and the app says
```
Unable to load script. Make sure you're either running a Metro server (run 'react-native start') or that your bundle 'index.android.bundle' is packaged correctly for release.
```
And if I do `react-native start` or `yarn start`, it starts the Metro bundler, starts listening on port 8081 and also serves the bundle ([see here](https://prnt.sc/ojqdsu)), but on the device, it is just blank([see here](https://prnt.sc/ojqeev)).

**Versions:**
Node : 12.4.0
react-native-cli: 2.0.1
react-native: 0.60.4

And I just noticed another weird thing while writing this comment.
That is if I do `react-native run-android` and the server fails to start,  then I do `yarn start`, the app stays blank even if I reload using the menu.
But if I exit the app and reopen it, everything works.([see here](https://prnt.sc/ojqnzr))
So, I think the problem is with the `react-native run-android` command.
Thanks for taking your time and replying, I am really hoping to be able to use v0.60.x but I am having to use 0.59.10 for my current projects.

**Comment by thymikee** [comment]
Hm you're on Linux. We've seen reports that Metro doesn't start properly there. What happens if you start fresh, run `start` and then `run-android`? If this works properly the issue is in `launchPackager` script that lives in RN core. Maybe you could debug that?

**Comment by Mahi-Uddin** [comment]
Yeah, it works if I do `yarn start` to start the server first, then `run-android`.
Everything works perfectly if I do that.

**Comment by thymikee** [comment]
If you'd like to help us finding a fix, this is the file that fails to do its job: https://github.com/facebook/react-native/blob/master/scripts/launchPackager.command

**Comment by Mahi-Uddin** [comment]
[This](https://github.com/facebook/react-native/blob/master/scripts/launchPackager.command) script executes 
[this](https://github.com/facebook/react-native/blob/master/scripts/packager.sh) one.
And in [this](https://github.com/facebook/react-native/blob/master/scripts/packager.sh#L24) line, it uses `$NODE_ENV` which is `node` when run the script directly.
But when I start it with `react-native run-android`, `$NODE_ENV` is empty.
This is where the problem is.
So, what I did for now is I added the following just before that line,
```sh
if [ -z "$NODE_BINARY" ]
then
  export NODE_BINARY="node"
fi
```
As you can see [here](https://github.com/Mahi-Uddin/react-native/blob/master/scripts/packager.sh#L24).
I am not sure if it is safe or okay to do, so I am not creating any pull request or anything.
But, let me know if I can.
I will be happy to.
Thanks.

**Comment by Mahi-Uddin** [comment]
I just noticed that [this](https://github.com/facebook/react-native/blob/master/scripts/node-binary.sh) script is never reached.
I tried to put a line for creating a file in my home directory at the very beginning of this script but nothing was created, and I assume that's why `NODE_BINARY` stays empty.

**Comment by Slaycaster** [comment]
I solved this problem when I'm trying to migrate from RN 0.55 to 0.60.

add a folder on `android\src` named `debug` and create an `AndroidManifest.xml` file there then add these:

```<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools">

    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>

    <application android:usesCleartextTraffic="true" tools:targetApi="28" tools:ignore="GoogleAppIndexingWarning" />
</manifest>```

then try to run `react-native run-android` again and it should listen to the Metro Bundler.

**Comment by thymikee** [comment]
We've recently released a new version of the CLI with a fix for such case. Can you please give it a try and let us know if it works for you? Here's how to [update the `@react-native-community/cli`](https://github.com/react-native-community/cli#updating-the-cli) to the latest version.

**Comment by Mahi-Uddin** [comment]
Same as before

**Comment by michaelmorar** [comment]
I have exactly Mahi-Uddin's problem. Running Ubuntu. Is downgrading to node 12.4.0 the only received solution?

**Comment by sayem314** [comment]
I had exact same issue as @Mahi-Uddin and after spending 2 days fixing I came to this thread via google and @thymikee saved me.

TLDR: Node 12 bad, Node 10 good :laughing:

**Comment by Mahi-Uddin** [comment]
> I had exact same issue as @Mahi-Uddin and after spending 2 days fixing I came to this thread via google and @thymikee saved me.
> 
> TLDR: Node 12 bad, Node 10 good

Did switching to node 10 also fixed the empty NODE_ENV issue?
Or you are on another platform?

**Comment by sayem314** [comment]
> > I had exact same issue as @Mahi-Uddin and after spending 2 days fixing I came to this thread via google and @thymikee saved me.
> > TLDR: Node 12 bad, Node 10 good
> 
> Did switching to node 10 also fixed the empty NODE_ENV issue?
> Or you are on another platform?

Not sure about that. I'm using both android and ios for projects. And the issue occurred with Ubuntu as well as macOS. Turns out I had upgraded node to 12 on the same day.

**Comment by michaelmorar** [comment]
Folks, I've downgraded to node v10 (and reinstalled the whole OS - long story) but had the same problem. run-android does not work. My emulator complains of no server running. 

I have used the suggested workaround of "yarn start" and then "run-android" - seems to work.

**Comment by thymikee** [comment]
@michaelmorar Please [update the `@react-native-community/cli`](https://github.com/react-native-community/cli#updating-the-cli) to the latest version.

**Comment by apsuva** [comment]
> @michaelmorar Please [update the `@react-native-community/cli`](https://github.com/react-native-community/cli#updating-the-cli) to the latest version.

@react-native-community/cli@2.8.3 , i have faced same issue

**Comment by Ade-Joshe** [comment]
I wouldn't like to think my issue is far fetched from all I've seen here.

What's new in mine is that the packager is running but it my device never finds it. It always flag unknown. 

Sometimes today, it ran successfully but  had issued with my USB and disconnected to never connect again.

**Comment by mhmdAljefri** [comment]
Same Issue

System:
    OS: Linux 4.15 Ubuntu 16.04.4 LTS (Xenial Xerus)
    CPU: (4) x64 Intel(R) Core(TM) i5-3230M CPU @ 2.60GHz
    Memory: 311.86 MB / 5.73 GB
    Shell: 5.1.1 - /usr/bin/zsh
  Binaries:
    Node: 10.15.0 - ~/.nvm/versions/node/v10.15.0/bin/node
    Yarn: 1.6.0 - /usr/bin/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v10.15.0/bin/npm
  SDKs:
    Android SDK:
      API Levels: 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 7, 8, 9
      Build Tools: 19.1.0, 20.0.0, 21.0.0, 21.0.1, 21.0.2, 21.1.0, 21.1.1, 21.1.2, 22.0.0, 22.0.1, 23.0.0, 23.0.1, 23.0.2, 23.0.3, 24.0.0, 24.0.1, 24.0.2, 24.0.3, 25.0.0, 25.0.1, 25.0.2, 25.0.3, 26.0.0, 26.0.2, 27.0.1, 27.0.3, 28.0.3
      System Images: android-10 | Intel x86 Atom, android-14 | ARM EABI v7a, android-15 | ARM EABI v7a, android-15 | MIPS, android-15 | Intel x86 Atom, android-16 | ARM EABI v7a, android-16 | MIPS, android-16 | Intel x86 Atom, android-17 | ARM EABI v7a, android-17 | MIPS, android-17 | Intel x86 Atom, android-18 | ARM EABI v7a, android-18 | Intel x86 Atom, android-19 | ARM EABI v7a, android-19 | Intel x86 Atom, android-20 | Android Wear ARM EABI v7a, android-20 | Android Wear Intel x86 Atom, android-21 | Android TV ARM EABI v7a, android-21 | Android TV Intel x86 Atom, android-21 | Android Wear ARM EABI v7a, android-21 | Android Wear Intel x86 Atom, android-21 | ARM EABI v7a, android-21 | Intel x86 Atom, android-21 | Intel x86 Atom_64, android-21 | Google APIs ARM EABI v7a, android-21 | Google APIs Intel x86 Atom, android-21 | Google APIs Intel x86 Atom_64, android-22 | Android TV ARM EABI v7a, android-22 | Android TV Intel x86 Atom, android-22 | Android Wear ARM EABI v7a, android-22 | Android Wear Intel x86 Atom, android-22 | ARM EABI v7a, android-22 | Intel x86 Atom, android-22 | Intel x86 Atom_64, android-22 | Google APIs ARM EABI v7a, android-22 | Google APIs Intel x86 Atom, android-22 | Google APIs Intel x86 Atom_64, android-23 | Android TV ARM EABI v7a, android-23 | Android TV Intel x86 Atom, android-23 | ARM EABI v7a, android-23 | Intel x86 Atom, android-23 | Intel x86 Atom_64, android-23 | Google APIs ARM EABI v7a, android-23 | Google APIs Intel x86 Atom, android-23 | Google APIs Intel x86 Atom_64, android-24 | ARM 64 v8a, android-25 | Android TV Intel x86 Atom, android-25 | Google APIs Intel x86 Atom, android-25 | Google APIs Intel x86 Atom_64, android-MNC | Android TV Intel x86 Atom, android-MNC | ARM 64 v8a, android-MNC | ARM EABI v7a, android-MNC | MIPS, android-MNC | Intel x86 Atom, android-MNC | Intel x86 Atom_64, android-N | Android TV Intel x86 Atom, android-N | Intel x86 Atom, android-N | Intel x86 Atom_64
  npmPackages:
    react: 16.8.6 => 16.8.6 
    react-native: 0.60.5 => 0.60.5 
  npmGlobalPackages:
    react-native-create-library: 3.1.2

**Comment by abhirup03** [comment]
I have upgraded my react-native app from RN 0.57 to RN 0.60.0. after react-native run-android the configurations successfully executes, but the metro bundler doesn't start. Can anyone help me fix this?

**Comment by Mahi-Uddin** [comment]
> I have upgraded my react-native app from RN 0.57 to RN 0.60.0. after react-native run-android the configurations successfully executes, but the metro bundler doesn't start. Can anyone help me fix this?

Start the bundler first with `yarn start` and then do `react-native run-android`.
If you see a blank screen with no errors on your phone, try restarting the app on your phone.
Hope it helps.

**Comment by AmitPandya007** [comment]
> > I have upgraded my react-native app from RN 0.57 to RN 0.60.0. after react-native run-android the configurations successfully executes, but the metro bundler doesn't start. Can anyone help me fix this?
> 
> Start the bundler first with `yarn start` and then do `react-native run-android`.
> If you see a blank screen with no errors on your phone, try restarting the app on your phone.
> Hope it helps.

I have tried the process in different system and it was running and JS loading is also visible. Any help on this one?

**Comment by matthewmazzuca** [comment]
I am also getting this issue after updating to 0.60. I noticed the following message in the Target Output in Xcode:

`nw_protocol_boringssl_get_output_frames(1301) [C11.1:2][0x7fc0716025f0] get output frames failed, state 8196`

**Comment by hohien** [comment]
same issue, is there any good solution?

**Comment by GaetanB999** [comment]
for me it's working fine when I do `react-native run-android` on Windows but not when I do `react-native run-ios` on MacOS. For this one I must do `yarn start` before.

for the two environment I work with Node: 10.16.3

**Comment by ivanzhang2** [comment]
i still reproduce this issue on mac, node v10, 
but only when run-ios , both RN 0.60.4 and 0.60.5 even on init project ...
but some of my old projects init with 0.60.3 was working  .... 
strange ...

**Comment by MakhouT** [comment]
I am also having this issue, I need to run `yarn start` manually , while before I could just do `react-native run-ios` and the bundler started automatically. 
Coming from 0.59.10
Currently on 0.60.5

**Comment by DirtySoc** [comment]
macos here. I was able to fix by running `yarn upgrade react-native-cli`

**Comment by thymikee** [comment]
@DirtySoc `yarn upgrade` upgrades _all_ packages, not the one you passed. FYI `react-native-cli` packages didn't get a release in 2+ years. We now use `@react-native-community/cli` under the hood of `react-native`.

**Comment by MakhouT** [comment]
@thymikee Is there then an alternative solution for this issue?

**Comment by thymikee** [comment]
Make sure to [update the `@react-native-community/cli`](https://github.com/react-native-community/cli#updating-the-cli) to the latest version, we've already fixed quite a bunch of bugs related to this issue already. Other than that you can always run `yarn react-native start` in a separate terminal tab.

**Comment by jacquesdev** [comment]
Will my react-native automatically use the new updated cli, or is there something I need to do to ensure it does this?

**Comment by MakhouT** [comment]
Seems I was already on `2.9.0`

**Comment by thymikee** [comment]
@MakhouT please create a new issue with a reproduction and environment data on the CLI repo: https://github.com/react-native-community/cli
@jacquesdev not always, because of how lock files work.

I'm gonna close this issue, as the original one is solved and now it's just a place for complaining that "it doesn't work on my machine", which is not really productive.

If you still experience this issue having the latest CLI installed, please file a new one at https://github.com/react-native-community/cli.

**Comment by badarshahzad** [comment]
Hi :wave:  facing with same issue 
node: v 8.12.0
npm : 6.4.1
react-native-cli: 2.0.1
react-native: 0.59.8

Linux 16.04

If someone solved this issue kindly tell me for last 15 hours I'm trying to start metro bundler to debug my application.

Thank you everyone for your discussion.

**Comment by badarshahzad** [comment]
> Hi  facing with same issue
> node: v 8.12.0
> npm : 6.4.1
> react-native-cli: 2.0.1
> react-native: 0.59.8
> 
> Linux 16.04
> 
> If someone solved this issue kindly tell me for last 15 hours I'm trying to start metro bundler to debug my application.
> 
> Thank you everyone for your discussion.

The metro bundler and debuger option appear for this version 
react-native:0.61.2
react-native-cli:2.0.1

**Comment by AmitPandya007** [comment]
@badarshahzad 
On which platform have you tried?
Android OR iOS?

**Comment by badarshahzad** [comment]
Android

**Comment by AmitPandya007** [comment]
in MainApplication.java file, for this below method if you are importing 
**import com.facebook.react.BuildConfig** package then remove it and run it again:

       ```
 @Override
        public boolean getUseDeveloperSupport() {
            return BuildConfig.DEBUG;
        }
```

**Comment by badarshahzad** [comment]
> @badarshahzad
> On which platform have you tried?
> Android OR iOS?

Android

**Comment by badarshahzad** [comment]
> **import com.facebook.react.BuildConfig**

I didn't find `com.facebook.react.BuildConfig` but I found this `import androidx.multidex.BuildConfig;`

**Comment by AmitPandya007** [comment]
Is multiDex needed for your project? If not then you can try removing it.

Is this method present in your **MainApplication.java** file, if not then please add it.
@override
public boolean getUseDeveloperSupport() {
return BuildConfig.DEBUG;
}

**Comment by AmitPandya007** [comment]
@badarshahzad Did it resolved your issue?

**Comment by MSandeep111** [comment]
@sayem314 thanks I used node 10.17 and it worked

**Comment by piyushgaurav07** [comment]
> Yeah, it works if I do `yarn start` to start the server first, then `run-android`.
> Everything works perfectly if I do that.

It worked for me too

**Comment by nicwhitts** [comment]
> I solved this problem when I'm trying to migrate from RN 0.55 to 0.60.
> 
> add a folder on `android\src` named `debug` and create an `AndroidManifest.xml` file there then add these:
> 
> ```
> <?xml version="1.0" encoding="utf-8"?>
> <manifest xmlns:android="http://schemas.android.com/apk/res/android"
>     xmlns:tools="http://schemas.android.com/tools">
> 
>     <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
> 
>     <application android:usesCleartextTraffic="true" tools:targetApi="28" tools:ignore="GoogleAppIndexingWarning" />
> </manifest>
> ```
> 
> then try to run `react-native run-android` again and it should listen to the Metro Bundler.

This worked for me!!!

I had this problem randomly, my device was running fine for months and one day would not connect

Thank you!

**Comment by lisichka999** [comment]
I am started migration from 59 to 61, currently only on IOS. I cannot see metro bundler at all, however the app installed properly on emulator, was it removed? I additional have some problem with debugging, may be it can be related? Thanks

**Comment by cavator** [comment]
2020 and this issue still going, i have no idea how get information like guys above, but it's happening like the first one says, bye and help, i can work with the start and run after, but i just want do things work normally, u know

**Comment by 1rjun** [comment]
Just remove 
**import com.facebook.react.BuildConfig;**
from MainApplication.java file.
Your problem will be resolved.

Cheers 👯‍♂

**Comment by cinder92** [comment]
> in MainApplication.java file, for this below method if you are importing
> **import com.facebook.react.BuildConfig** package then remove it and run the metro builder again:
> 
> > @override
> > public boolean getUseDeveloperSupport() {
> > return BuildConfig.DEBUG;
> > }

whhaaaaaaat? why this works now? Thanks a lot!

**Comment by archfz** [comment]
Related hotfix for linux in  #28807

**Comment by prabhuraj94** [comment]
> in MainApplication.java file, for this below method if you are importing
> **import com.facebook.react.BuildConfig** package then remove it and run the metro builder again:
> 
> > @override
> > public boolean getUseDeveloperSupport() {
> > return BuildConfig.DEBUG;
> > }

@AmitPandya007 You're a life saver :D but how does this work?

**Comment by whatdtech** [comment]
Reinstalling android studio with a new emulator or changing the app name in app.json , build.gradle and android manifest.xml solved the issue.

**Comment by airtonix** [comment]
I find it absolutely astounding that the react native cli touches and makes assumptions about what the nodejs binary will be.

thoroughly disgusted.

I imagine this mindset then led to further hacks like making assumptions about whether someone is using nvm or not.

**Comment by thymikee** [comment]
@airtonix the code is open, feel free to fix it then :)

**Comment by mppvgithub** [comment]
> in MainApplication.java file, for this below method if you are importing
> **import com.facebook.react.BuildConfig** package then remove it and run the metro builder again:
> 
> > @override
> > public boolean getUseDeveloperSupport() {
> > return BuildConfig.DEBUG;
> > }

Even I try this, does not work!

**Comment by lalitkumar50678** [comment]
Hii ,

I found this issue in my current project in iOS it's working fine in android . The project is developed in react-native 0.66.0 . 
I try from last 10 days but nothing  is working in google i read all most all article related to this . After  i search how the internal thing is working how react-native connect with bundle. 

After 10 min of R&D i found that in AppDelegate.m there is one function 
 (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
 
 `- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
 return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}
`


When put the break point here i found that in debug mode it's check release part. So for now comment the release code. 
Now my this function look like this 

`- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
//#if DEBUG
//  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
//#else
//  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
//#endif
}`


Now my code this working .
